### PR TITLE
Snowflake ALTER TABLE: Add clusteringAction

### PR DESCRIPTION
**Pull request by jpy-git**
﻿

### Brief summary of the change made

Adds `clusteringAction` to Snowflake `ALTER TABLE`.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -1040,6 +1040,52 @@ class AlterTableTableColumnActionSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class AlterTableClusteringActionSegment(BaseSegment):
+    """ALTER TABLE `clusteringAction` per defined in Snowflake's grammar.
+
+    https://docs.snowflake.com/en/sql-reference/sql/alter-table.html#clustering-actions-clusteringaction
+
+    If possible, please match the order of this sequence with what's defined in Snowflake's
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_snowflake.py:1049]: The wording of this sentence seems strange. Why "please match"?

**Review by barrywhart — APPROVED** [review]
Code looks good. Question about wording of docstring.

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_snowflake.py:1060]: I just copied the phrase from the similar `AlterTableTableColumnActionSegment` introduced [here](https://github.com/sqlfluff/sqlfluff/pull/2267#discussion_r781325367) 🤷

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_snowflake.py:1060]: tbh I think it's redundant anyway so just gonna remove it 👍

**Pull request merged**